api/v1beta1/userinfo.go

```diff
@@ -72,3 +76,80 @@ func setUserInfo(userInfo authV1.UserInfo, obj metaV1.Object) error {
 
 	return nil
 }
+
+func filterUserGroupsByPermitted(userInfo authV1.UserInfo, permittedGroups map[string]struct{}) []string {
+	var filteredGroups []string
+	for _, group := range userInfo.Groups {
+		if _, ok := permittedGroups[group]; ok {
+			filteredGroups = append(filteredGroups, group)
+		}
+	}
+
+	return filteredGroups
+}
+
+func getUserInfoFromObject(object metaV1.Object) (authV1.UserInfo, error) {
+	switch d := object.(type) {
+	case *Disruption:
+		return d.UserInfo()
+	case *DisruptionCron:
+		return d.UserInfo()
+	}
+
+	return authV1.UserInfo{}, fmt.Errorf("unable to extract user info from object")
+}
+
+// validateUserInfoImmutable checks that no changes have been made to the oldDisruption's UserInfo in the latest update
+func validateUserInfoImmutable(oldObject, newObject client.Object) error {
+	oldUserInfo, err := getUserInfoFromObject(oldObject)
+	if err != nil {
+		return nil
+	}
+
+	emptyUserInfo := fmt.Sprintf("%v", v1beta1.UserInfo{})
+	if fmt.Sprintf("%v", oldUserInfo) == emptyUserInfo {
+		return nil
+	}
+
+	userInfo, err := getUserInfoFromObject(newObject)
+	if err != nil {
+		return err
+	}
+
+	if fmt.Sprintf("%v", userInfo) != fmt.Sprintf("%v", oldUserInfo) {
+		return fmt.Errorf("the user info annotation is immutable")
+	}
+
+	return nil
+}
+
+// validateUserInfoGroup checks that if permittedUserGroups is set, which is controlled in controller.safeMode.permittedUserGroups in the configmap,
+// then we will return an error if the user in r.UserInfo does not belong to any groups. If permittedUserGroups is unset, or if the user belongs to one of those
```

Comment:
```suggestion
// then we will return an error if the user in r.UserInfo does not belong to any of the permitted groups. If permittedUserGroups is unset, or if the user belongs to one of those
```

Looks like the old comment was ambiguous